### PR TITLE
RequestMoney API

### PR DESCRIPTION
**Pull request by VipinDevineni**

### CODE REVIEW

File: application/api/request_service.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                        amount = amt,
+                        status = RequestMoney.PENDING,
+                        payment_date = due_date,
+                        memo = None,
+                        time_updated = now,
+                        time_created = now)
+                    account.request_money_list.append(req)
+                    self.db_session.add(account)
+                    self.db_session.add(req)
+                    self.db_session.flush()
```
Comment (susnata1981) [inline application/api/request_service.py:128]: I thought commit will automatically flush everything

File: application/db/model.py [hunk 2]
```diff
@@ -203,6 +222,21 @@ class RequestMoney(Base):
     time_updated = Column(DateTime)
     time_created = Column(DateTime)
 
+class RequestMoneyHistory(Base):
+    __tablename__ = "request_money_history"
+
+    id = Column(Integer, ForeignKey('request_money.id'))
```
Comment (susnata1981) [inline application/db/model.py:237]: Don't we want  auto_increment primary key and a request_id foreign key?

File: application/db/model.py [hunk 3]
```diff
@@ -235,13 +285,30 @@ class Transaction(Base):
...
+class TransactionHistory(Base):
+    __tablename__ = 'transaction_history'
+
+    id = Column(Integer, ForeignKey('transaction.id'))
+    transaction = relationship('Transaction', uselist=False, back_populates="history")
+    status = Column(Integer, nullable=False)
+    memo = Column(Text, nullable=True)
+    time_created = Column(DateTime)
+    __table_args__ = (
+        PrimaryKeyConstraint('id', 'time_created'),
```
Comment (susnata1981) [inline application/db/model.py:306]: primary key based on timestamp, is that a good idea?

File: application/db/model.py [hunk 4]
```diff
@@ -203,6 +222,21 @@ class RequestMoney(Base):
     time_updated = Column(DateTime)
     time_created = Column(DateTime)
 
+class RequestMoneyHistory(Base):
+    __tablename__ = "request_money_history"
+
+    id = Column(Integer, ForeignKey('request_money.id'))
+    request = relationship('RequestMoney', uselist=False, back_populates="history")
+    amount = Column(Float, nullable=False)
```
Comment (susnata1981) [inline application/db/model.py:239]: can request amount change?

**Review by susnata1981 — COMMENTED** [review]
left couple of questions.

File: application/db/model.py (hunk 2 above)
Comment (VipinDevineni) [reply application/db/model.py:237]: The 'history' id is a foreign key of the original.

File: application/api/request_service.py (hunk 1 above)
Comment (VipinDevineni) [reply application/api/request_service.py:128]: Need the `id` so I can add it to the history object. can get this only by flush

File: application/db/model.py (hunk 4 above)
Comment (VipinDevineni) [reply application/db/model.py:239]: thought about it, added it just in case we support amount change - approve them for less amount with a memo ....

**Pull request merged**